lightning/src/ln/channelmanager.rs

```diff
@@ -3131,6 +3158,90 @@ impl<M: Deref, T: Deref, K: Deref, F: Deref, L: Deref> ChannelManager<M, T, K, F
 		Ok(())
 	}
 
+	/// Attempts to forward an intercepted HTLC over the provided scid and with the provided
+	/// amount to forward. Should only be called in response to an [`HTLCIntercepted`] event.
+	///
+	/// Intercepted HTLCs can be useful for Lightning Service Providers (LSPs) to open a just-in-time
+	/// channel to a receiving node if the node lacks sufficient inbound liquidity.
+	///
+	/// To make use of intercepted HTLCs, use [`ChannelManager::get_intercept_scid`] to generate short
+	/// channel id(s) to put in the receiver's invoice route hints. These route hints will signal to
+	/// LDK to generate an [`HTLCIntercepted`] event when it receives the forwarded HTLC, and this
+	/// method or [`ChannelManager::fail_intercepted_htlc`] MUST be called in response to the event.
+	///
+	/// Note that LDK does not enforce fee requirements in `amt_to_forward_msat`.
```

Comment:
If we would like just to outright reject it, I think one approach could be to extend `PendignAddHTLCInfo` with `prev_inbound_amount_msat`. Otherwise, we can just modify the documentation the user to warn about the over-forwarding concern and put the responsibility on them.

Reply:
Expanded the docs to warn that we won't prevent over-forwarding